### DISCUSSION TITLE
Is there any problem with relative paths in Quarkus native ?

### DISCUSSION THREAD

**Discussion by nicolasduminil** [quarkusio/quarkus#33796] [Community]
Hi,

I'm compiling in native mode a Quarkus application that works as expected in JVM mode. It happens to be a Camel application which one of the routes looks like:

       ...
       fromF("file://%s?include=.*.xml&delete=true&idempotent=true&bridgeErrorHandler=true", inBox)
         .doTry()
           .to("validator:xsd/money-transfers.xsd")
       ...
This route is polling an input folder and, as soon as an XML file lands in it, it tries to validate it against the schema named `money-transfers.xsd`, found in the `xsd` directory. This directory is located in `src/main/resources` folder of the maven project and, after the build, it lands at the root of the resulting JAR.

This works as expected in JVM mode but, trying to run it in native mode raises the following exception:

    2023-06-02 14:18:48,788 ERROR [org.apa.cam.imp.eng.AbstractCamelContext] (main) Error starting CamelContext (camel-1) due to exception thrown: Failed to start route route1 because of null: org.apache.camel.FailedToStartRouteException: Failed to start route route1 because of null
    ...
    Caused by: org.apache.camel.RuntimeCamelException: java.io.FileNotFoundException: Cannot find resource: xsd/money-transfer.xsd for URI: xsd/money-transfer.xsd
        at  org.apache.camel.component.validator.DefaultLSResourceResolver$DefaultLSInput.getByteStream(DefaultLSResourceResolver.java:139)

In the JAR associated to the native image I have:

    $ jar tvf target/aws-camelk-file-native-image-source-jar/aws-camelk-file-runner.jar
      ...
          0 Fri Jun 02 14:16:44 CEST 2023 xsd/
          890 Fri Jun 02 14:16:44 CEST 2023 xsd/money-transfers.xsd
         2118 Fri Jun 02 14:16:44 CEST 2023 xsd/money-transfer.xsd
    ...

Is there any reason not to be able to locate the mentioned file ?

Many thanks in advance.

Nicolas

**Comment by maxandersen** [comment]
My guess is that somewhere in the camel stack the code might assume the classpath is accessible as a file system. That is not the case in native but also shouldn't be assumed in jvm mode. 

Are you able to make a reproducer ?

@ppalaga might have more of an idea of what causes it.

**Reply by nicolasduminil** [reply]
Sure, [here](https://github.com/nicolasduminil/aws-camelk/tree/native/aws-camelk-file)

**Comment by ppalaga** [comment]
You are including xsd/money-transfers.xsd in application.properties via

```
quarkus.native.resources.includes=xsd/money-transfers.xsd
```

but the error message complains about xsd/money-transfer.xsd without `s`.

So you perhaps want to include all XSDs?

```
quarkus.native.resources.includes=xsd/*.xsd
```

**Comment by nicolasduminil** [comment]
This was it, many thanks.
